app/controllers/sessions_controller.rb

```diff
@@ -0,0 +1,25 @@
+class SessionsController < ApplicationController
+    get '/sessions/new' do
+        erb :"sessions/new.html"
+    end
+
+    post '/sessions' do
+        user = User.find_by(email: params[:email])
+        if user.authenticate(params[:password])
+            flash[:notice] = "You are logged in as " + params[:email]
+            session[:user_id] = user.id
+	        redirect "/"
+        else 
+            @email = params[:email]
+            @error = "Invalid email or password"
```

Comment:
Rather than storing the email and errors as instance variables, you can pass them into the `erb` template via. `locals`